### DISCUSSION TITLE
[BUG] rofi 1.7.0 ignores -lines argument

### DISCUSSION THREAD

**Discussion by yamirui** [davatorium/rofi#1447] [Theming] [answered]
### Rofi version (rofi -v)

Version: 1.7.0

### Configuration

https://gist.github.com/yamirui/e7047ea51b421931931f0f4241ff59f0

### Launch command

rofi -show window -lines 4

### Step to reproduce

Running the command is enough to observe the problem.

Alternatives that also exhibit identical behaviour:

`rofi -lines 4 -show window`
`rofi -show window -l 4`
`rofi -l 4 -show window`
`rofi -l 9999 -show window`
`rofi -no-default-config -show drun -l 4`

(option seems to be outright ignored, in ALL of modes I've tried)

### Expected behavior

I expect rofi to show only 4 line high block, centered on my screen, no more, no less, and produce warnings/errors if argument I supplied is not one supported by rofi.

### Actual behavior

It seems to ignore the option completely and shows whatever the default is, which is more than 4 lines, it covers almost the entire screen in height.
rofi produces 0 warnings or errors about this.

EDIT: `-no-fixed-num-lines` "fixes" this. It works with `-lines 0`, displaying 0 extra lines besides the input box as expected; does not with `rofi -no-fixed-num-lines -show drun -lines 4` where input box is now centered like it knows that only 4 lines should be displayed, but it grows past the bottom of the screen, and also shrinks beyond 4 lines as the input matches less than 4 results.
 
### Additional information

EDIT: This behaviour may be intended, but it is confusing. I'd expect it to always give priority to `-lines N` argument without caring about the `fixed-num-lines`. I'm not really sure what it is for or why it should affect this argument either, it seems to have a different purpose than allowing `-lines` to limit amount of lines displayed, when a flag saying "no limited  number of lines" is enabled...? Even after this I believe that is a bug.

Take note that provided config and theme is the default for installation on my distro, and I can reproduce this without changing anything.
I am using Arch Linux, but I doubt its the issue introduced by package maintainers.
`-lines` option used to work before this update the way I expected it to, it seems that it was renamed to `-l` (can't find longer option -lines in help anymore) but that option is equally ignored.
I can't think of a way how a theme could override this option and "break" it in intended way so I believe that this IS indeed a bug, whether in the default theme, or rofi itself.

**Comment by DaveDavenport** [comment, marked as answer]
Please see the [sticky](https://github.com/davatorium/rofi/issues/1397).   The `-lines`  option has been discussed several times in the discussion form.

**Reply by DaveDavenport** [reply]
for example https://github.com/davatorium/rofi/discussions/1407

**Reply by jneidel** [reply]
> for example #1407

I don't want to set this in a config file, I want to set it in my script depending on context.
I feel like this is removing a useful CLI option.

**Reply by DaveDavenport** [reply]
There is a solution in the works that can add  supported by the theme for this. This is also discussed here: https://github.com/davatorium/rofi/discussions/1455, https://github.com/davatorium/rofi/discussions/1411.

**Reply by jneidel** [reply]
And having [this kind of weird syntax](https://github.com/davatorium/rofi/discussions/1455) for passing `-lines` is so unintuitive, especially as I can use `-l` with `-dmenu` in some scripts, but not all.

**Reply by DaveDavenport** [reply]
I have no idea where you point too, you point to the main post, not a specific item.

In my opinion, how rofi looks belongs to the theme.  Modifying this via commandline options (and we would need a lot of them to support all theme options)  is likely to break stuff.
This has happened many times in the past. That is why I tried to separate the two further.

There is going to be a way that a theme can add specific support for modifying some of it properties via  environment variable in the theme itself.  This should be very generic and therefor easier to maintain.

I am sorry if you don't like this, but I cannot please everybody. I need to keep rofi maintainable in the limited development time I have for it and streamlining code by removing tons of legacy options was needed.


( I don't understand your dmenu and `-l` remark and it sometimes working and sometimes not)

**Reply by yamirui** [reply]
Thank you.

@DaveDavenport it was pretty confusing to me how it worked at the time I wrote this. Essentially, that option really changes nothing, the `-l` is "broken". And doesn't change anything. That particular instance looked to me like it worked... but that's because I had 4 different windows open. Maybe I should undo the edits to make it less confusing then?

Also, I agree with everything here, how rofi looks should be delegated to the theme, not the commandline options.

I was relying on `-lines` option to limit the lines (excluding the input area) displayed at once to 4, but in some cases I was using it to disable any lines shown at all (`-lines 0`). But if a theme can essentially do the same thing of limiting entries up to 4 while growing downwards from 0 to 4 otherwise, I'm okay with it.
